src/vl-map-layer-switcher.js

```diff
@@ -0,0 +1,88 @@
+import {vlElement} from '/node_modules/vl-ui-core/dist/vl-core.js';
+import '/node_modules/vl-ui-form-message/dist/vl-form-message.js';
+
+/**
+ * VlMapLayerSwitcher
+ * @class
+ * @classdesc De kaartlagen wisselaar.
+ *
+ * @extends HTMLElement
+ * @mixes vlElement
+ *
+ * @property {boolean} [data-vl-title=Kaartlagen] - Attribute wordt gebruikt om de kaartlagen titel te bepalen.
+ *
+ * @see {@link https://www.github.com/milieuinfo/webcomponent-vl-ui-map/releases/latest|Release notes}
+ * @see {@link https://www.github.com/milieuinfo/webcomponent-vl-ui-map/issues|Issues}
+ * @see {@link https://webcomponenten.omgeving.vlaanderen.be/demo/vl-map-overview-map.html|Demo}
+ */
+export class VlMapLayerSwitcher extends vlElement(HTMLElement) {
+  constructor() {
+    super(`
+      <style>
+        @import '/node_modules/vl-ui-form-message/dist/style.css';
+
+        :host {
+          display: block;
+        }
+
+        label {
+          display: block;
+        }
+      </style>
+      <div>
+        <label is="vl-form-label">Kaartlagen</label>
+        <slot></slot>
+      </div>
+    `);
+  }
+
+  connectedCallback() {
+    this._addClickListeners();
+  }
+
+  get _slot() {
+    return this._element.querySelector('slot');
+  }
+
+  get _layerInputs() {
+    return this._slot.assignedElements().filter((input) => input.hasAttribute('data-vl-layer'));
+  }
+
+  get _mapElement() {
+    return this.closest('vl-map');
+  }
+
+  get _map() {
+    if (this._mapElement) {
+      return this._mapElement._map;
+    }
+  }
+
+  _addClickListeners() {
+    if (this._map) {
+      this._layerInputs.forEach((input) => {
+        this._initializeInput(input);
+        input.addEventListener('click', () => this._setLayerVisibility(input));
+      });
+    }
+  }
+
+  _initializeInput(input) {
+    const layer = this._getLayer(input);
+    if (layer) {
+      input.checked = layer.getVisible();
+    }
+  }
+
+  _setLayerVisibility(input) {
+    const layer = this._getLayer(input);
+    if (layer) {
+      layer.setVisible(input.checked);
+      this._map.render();
```

Comment:
nodig om heel de map te rerenderen? alleen de layer kan niet?

Reply:
Inderdaad, feature en kaartlaag opnieuw renderen bleek niet voldoende.

Reply:
raar

Reply:
Ik vond het op zich ook raar, heb het tijdens ontwikkeling ook verder onderzocht maar helaas geen antwoord gevonden. De enige piste die me logisch leek was dat we gebruik maken van een `styleFunction` maar dit heb ik niet geverifieerd via testen omdat de huidige oplossing niet nadelig lijkt voor de performantie.

Voor de volledigheid heb ik issue #175 aangemaakt zodat we hier later nog op kunnen focussen.